server/src/main/java/org/opensearch/action/admin/cluster/state/term/TransportGetTermVersionAction.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.state.term;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.clustermanager.TransportClusterManagerNodeReadAction;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.block.ClusterBlockException;
+import org.opensearch.cluster.coordination.ClusterStateTermVersion;
+import org.opensearch.cluster.metadata.IndexNameExpressionResolver;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportService;
+
+import java.io.IOException;
+
+/**
+ * Transport action for obtaining cluster term and version from cluster-manager
+ *
+ * @opensearch.internal
+ */
+public class TransportGetTermVersionAction extends TransportClusterManagerNodeReadAction<GetTermVersionRequest, GetTermVersionResponse> {
```

Comment:
How is the performance impact of this action being measured, what is the improvement?

Reply:
From the tests on 500 node cluster with 450K shards the following are the latency improvements.
API : `/cluster/_state`
Cluster State Size -  165 MB

With TermVersion check  :- 
P50 Latency: 4.7471 seconds
P90 Latency: 5.5958 seconds

Without TermVersion check :-
P50 Latency: 6.3810 seconds
P90 Latency: 7.3598 seconds

Reply:
I would also like to run this test case, how can I do that?  Lets get this test included in this PR.

Reply:
Added a test(`FetchByTermVersionIT`)to fetch large cluster state with & w/o term check. The test demonstrates the difference in latency of ClusterState API.

Reply:
`20 Index with each index having 5 shards and 10K mappings `
```
[2024-03-19T00:17:03,132][INFO ][o.o.c.s.FetchByTermVersionIT] [testClusterStateResponseFromDataNode] cluster().state() fetch with Term Version enabled took 1 milliseconds
[2024-03-19T00:17:04,710][INFO ][o.o.c.s.FetchByTermVersionIT] [testClusterStateResponseFromDataNode] cluster().state() fetch with Term Version disabled took 19 milliseconds
```

Reply:
@rajiv-kv Thanks for including these tests - great way to validate the expectation of this change and allow for tweaking if any issues come up.